src/swap/types.ts

```diff
@@ -107,13 +109,40 @@ export type LifeCycleStatus =
       statusName: 'amountChange';
       statusData: null;
     }
+  | {
+      statusName: 'transactionApproved';
+      statusData: {
+        transactionHash: Address;
+      };
+    }
+  | {
+      statusName: 'transactionPermit'; // Need to triple check this
+      statusData: {
+        transactionPermit: Address;
+      };
+    }
+  | {
+      statusName: 'transactionPermitApproved'; // Need to triple check this
+      statusData: {
+        transactionHash: Address;
+      };
+    }
```

Comment:
3 states for swap - `erc20Approved`, `permit2Approved`, and `success`

for V2 API, it goes `permit2Approved` -> `erc20Approved` -> `success`
for V1 API, it goes `erc20Approved` -> `success`

```suggestion
  | {
      statusName: 'erc20Approved';
      statusData: {
        transactionHash: Hex;
      };
    }
  | {
      statusName: 'permit2Approved';
      statusData: {
        transactionHash: Hex;
      };
    }
```